internal/provider/resource_group.go

```diff
@@ -0,0 +1,308 @@
+package provider
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/aws/smithy-go/ptr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	tharsis "gitlab.com/infor-cloud/martian-cloud/tharsis/tharsis-sdk-go/pkg"
+	ttypes "gitlab.com/infor-cloud/martian-cloud/tharsis/tharsis-sdk-go/pkg/types"
+)
+
+// GroupModel is the model for a group.
+type GroupModel struct {
+	ID          types.String `tfsdk:"id"`
+	Name        types.String `tfsdk:"name"`
+	Description types.String `tfsdk:"description"`
+	ParentPath  types.String `tfsdk:"parent_path"`
+	FullPath    types.String `tfsdk:"full_path"`
+	LastUpdated types.String `tfsdk:"last_updated"`
+}
+
+// Ensure provider defined types fully satisfy framework interfaces
+var (
+	_ resource.Resource                = &groupResource{}
+	_ resource.ResourceWithConfigure   = &groupResource{}
+	_ resource.ResourceWithImportState = &groupResource{}
+)
```

Comment:
Instead of assigning a value and disregarding it, we can use a nil pointer to the type.

```suggestion
var (
	_ resource.Resource                = (*groupResource)(nil)
	_ resource.ResourceWithConfigure   = (*groupResource)(nil)
	_ resource.ResourceWithImportState = (*groupResource)(nil)
)
```